src/crimson/osd/backfill_state.cc

```diff
@@ -394,6 +394,7 @@ boost::statechart::result
 BackfillState::PrimaryScanning::react(PrimaryScanned evt)
 {
   logger().debug("{}", __func__);
+  evt.result.version = backfill_state().backfill_info.version;
```

Comment:
Continuing my previous comment, I think that this will be correct without this line.

Reply:
We are not capturing `peering_state.info.last_update` here, at this moment `backfill_state().backfill_info.version` is the last_update captured in `PrimaryScanning::PrimaryScanning(my_context ctx)`

Reply:
Without this line, `backfill_state().backfill_info.version` will be overwritten with the empty value.

Reply:
Got it, approving. I'll add a comment to this section in my refactor to make it more explicit. I plan to require passing version once populating the Interval. Thanks for explaining

Reply:
@Matan-B Sorry, I think I was wrong and you were correct, we should only set the primary osd's backfill_info.version after we've done "update_range" when we can be sure we have all objects before `projected_last_update`, there's no point for us to set it before "update_range". @athanatos @Matan-B Am I right? If so, I'll close this PR.

Reply:
@Matan-B This line is wrong, you're right.

Reply:
@athanatos @Matan-B Maybe always setting `primary_bi.version` to `last_complete_ondisk` can solve this issue, what do you think?

Reply:
>1. An IO request set `projected_last_update` to `100'200` and creates object `obj_k`;
>2. before the IO request is committed, a primary scan happens, adds all objects from `obj_a to obj_h` to the backfill set and set `primary_bi.version` to `100'200`
>3. also before the IO request is committed, another round of primary scan happens and adds all objects from `obj_i to obj_m` to the backfill set, at which point, `obj_k` won't be in the backfill set because the IO request is not committed and `primary_bi.version` is already at `100'200`
>4. `obj_k` won't be backfilled eventually.

Once  `projected_last_update` is set to `100'200` (step 1), and `primary_bi.version` is set to `100'200` (step2) - that means that the `projected_log` will include this all updates/deletes (even uncommitted ones) up to this version.
So that later on, in step 3, before starting the pushes, we will update the backfill interval based on the projected log entries.
Can we use this mechanism also for being able to update (uncommitted) new objects such as `obj_k` above?

Reply:
The backfill_interval version should always be set to the last_update_ondisk on the osd where the scan was performed at the time when it started because last_update_ondisk is, definitionally, the version such that all prior updates will be caught by the scan.

Reply:
While progressing through the interval, the primary's interval should then be caught up using the projected log.